src/OFS/DTMLMethod.py

```diff
@@ -266,10 +266,12 @@ def manage_upload(self, file='', REQUEST=None):
         if self.wl_isLocked():
             raise ResourceLockedError('This DTML Method is locked.')
 
-        if not isinstance(file, binary_type):
+        if not isinstance(file, basestring):
             if REQUEST and not file:
                 raise ValueError('No file specified')
             file = file.read()
+            if PY3:
```

Comment:
Shouldn't the conversion binary_type->text_type happen for Python 2 as well? I would assume the file type is always binary_type (in which case the method's default should also be changed to `file=b''`).

Reply:
@rbu I think you are generally right. But there are databases out in the world which have stored the file contents as bytes (aka Python 2 `str`). Can we provide a way to convert their file contents to text so we do not have to live forever with file contents wich might be `bytes`?

Reply:
Ok I see what you're saying. In the old code, Python 2 would always have `str` byte strings and byte file objects. Those were read, resulting in bytes being saved to ZODB. As I understand it, per the WSGI spec, `file` will usually be a bytes file object, so `read()`ing from it needs to be decoded. So the goal of this code is to always save byte strings in PY2 and always save text strings in PY3.

To make that consistent, though, the default method parameter needs to be decoded as well. What do you think of the following. What do you think of this:
```

        if not isinstance(file, basestring):
            if REQUEST and not file:
                raise ValueError('No file specified')
            file = file.read()

        if PY3 and isinstance(file, binary_type):
                file = file.decode('utf-8')
 ```

Reply:
You are right if `manage_upload` is called with a `file` parameter which is `bytes` it should be decoded, too.